demisto_sdk/commands/download/downloader.py

```diff
@@ -1223,19 +1223,25 @@ def update_data(
             splitter="dot",
         )
 
-        if file_ending == "yml":
-            file_yaml_object = get_yaml(file_path_to_write)
-            if pack_obj_data:
-                merge(file_yaml_object, preserved_data)
-            with open(file_path_to_write, "w") as yf:
-                yaml.dump(file_yaml_object, yf)
-
-        elif file_ending == "json":
-            file_data: dict = get_json(file_path_to_write)
-            if pack_obj_data:
-                merge(file_data, preserved_data)
-            with open(file_path_to_write, "w") as jf:
-                json.dump(data=file_data, fp=jf, indent=4)
+        if file_ending not in {"yml", "json"}:
+            logging.warning(f"trying to merge an non-supported file {output_path}")
+            return
+
+        file_data = get_file(output_path, type_of_file=file_ending, clear_cache=True)
```

Comment:
why clear cache?

Reply:
to make sure we get the updated file, rather than an irrelevant version. We're editing files here